app/controllers/books_controller.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+class BooksController < ApplicationController
+  BOOKS_ON_PAGE = 12
+
+  include Pagy::Backend
+
+  include Filtering
+
+  decorates_assigned :books, :book
+
+  def index
+    @books_count = Book.all.count
+    scoped_books = Books::FindBooks.new(params).call
+    Pagy::VARS[:max_items] = BOOKS_ON_PAGE
+    @pagy, @books = pagy_countless(scoped_books, items: BOOKS_ON_PAGE, link_extra: 'data-remote="true"')
```

Comment:
Do you redefine here items count?
```suggestion
    @pagy, @books = pagy_countless(scoped_books, link_extra: 'data-remote="true"')
```